Doc: release_note_1.5 (#4328)
projectacrn/acrn-hypervisor: doc/release_notes/release_notes_1.5.rst

.. _release_notes_1.5:

ACRN v1.5 (Jan 2020)
####################

We are pleased to announce the release of ACRN version 1.5.

ACRN is a flexible, lightweight reference hypervisor that is built with
real-time and safety-criticality in mind. It is optimized to streamline embedded
development through an open source platform. Check out the :ref:`introduction` for more information.
All project ACRN source code is maintained in the https://github.com/projectacrn/acrn-hypervisor
repository and includes folders for the ACRN hypervisor, the ACRN device
model, tools, and documentation. You can either download this source code as
a zip or tar.gz file (see the `ACRN v1.5 GitHub release page
<https://github.com/projectacrn/acrn-hypervisor/releases/tag/v1.5>`_)
or use Git clone and checkout commands::

   git clone https://github.com/projectacrn/acrn-hypervisor
   cd acrn-hypervisor
   git checkout v1.5

The project's online technical documentation is also tagged to correspond
with a specific release: generated v1.5 documents can be found at https://projectacrn.github.io/1.5/.
Documentation for the latest (master) branch is found at https://projectacrn.github.io/latest/.

> ACRN v1.5 requires Clear Linux* OS version 31XXX. Follow the

Reviewer: Please complete version 31XXX.

Author: Waiting for final clear linux version.

instructions in the :ref:`rt_industry_setup`.

Version 1.5 major features
**************************

> What's New in v1.5

Reviewer: This is v1.4. Please delete.

==================
* ACRN now conforms to the Microsoft* Hypervisor Top-Level Functional Specification (TLFS).
* ACRN scheduler framework re-architected capabilities have been added.
* WaaG (Windows as a guest) stability and performance has been improved.
* Realtime performance of the RTVM (preempt-RT kernel-based) has been improved.

Document updates
================

> Many new `reference documents <https://projectacrn.github.io>`_ are available, including:

Reviewer: All of these documents were new in v1.4. Please remove.

Author: Will update for v1.5 release document.

* :ref:`ACRN high-level design <hld>` documents.
* :ref:`enable-s5`
* :ref:`enable_laag_secure_boot`
* :ref:`How-to-enable-secure-boot-for-windows`
* :ref:`asa`

> Security Vulnerabilities

Reviewer: @NanlinXie The "Security Vulnerabilities" section was included in v1.4. Does it need to be included again?

   not expected by the rest of the application. This can lead to parts of the system
   receiving unintended inputs, which may result in an altered control flow, arbitrary control
   of a resource, or arbitrary code execution.

FILE Pointer Is Not Closed After Using
   This vulnerability was reported by the Fuzzing tool. Leaving the file unclosed will cause a
   leaking file descriptor and may cause unexpected errors in the Device Model program.

Descriptor of Directory Stream Is Referenced After Release
   This vulnerability was reported by the Fuzzing tool. A successful call to ``closedir(DIR *dirp)``
   also closes the underlying file descriptor associated with ``dirp``. Access to the released
   descriptor may point to some arbitrary memory location or cause undefined behavior.

Mutex Is Potentially Kept in a Locked State Forever
   This vulnerability was reported by the Fuzzing tool. Here, ``pthread_mutex_lock/unlock`` pairing was not
   always done. Leaving a mutex in a locked state forever can cause program deadlock,
   depending on the usage scenario.

We recommend that all developers upgrade to ACRN release v1.4.

Version 1.5 major features
********************

> What’s New in v1.5

Reviewer: This "What's New" section can benefit from links to specific commits or web pages. Also, I edited your statements to make the tenses consistent.

********************

> Basic CPU sharing: Fairness Round-Robin CPU Scheduler was added to support basic CPU sharing (Service VM and WaaG share one CPU core)

Reviewer: ```suggestion
Basic CPU sharing: Fairness Round-Robin CPU Scheduler has been added to support basic CPU sharing (Service VM and WaaG share one CPU core).
```

> 8th gen Intel® Core ™ Processors (code name Whiskey Lake) are supported and validated

Reviewer: ```suggestion
8th Gen Intel® Core ™ Processors (code name Whiskey Lake) are now supported and validated.
```

> Further improved stability

Reviewer: ```suggestion
Further improved stability.
```

Reviewer: This is a very broad statement. What kind of stability? Security? Speed? Firmware?

> Offline configuration tool to ease developers porting ACRN to different hardware boards.

Reviewer: ```suggestion
Offline configuration tool has been created to ease developers porting ACRN to different hardware boards.
```

Author: Done!

- :acrn-issue: '4212' -[KBL][acrn-configuration-tool][WaaG+RTVM]Need support pm_channel&pm_by_vuart setting for Board:nuc7i7dnb+WaaG&RTVM
- :acrn-issue: '4227' -[ISD][Stability][WaaG][Regression] "Passmark8.0-Graphics3D-DirectX9Complex" test failed on WaaG due to driver error
- :acrn-issue: '4228' -[acrn-configuration-tool] cannot boot hypervisor on customer board with KBL 7300U
- :acrn-issue: '4229' -Add range check in Kconfig.
- :acrn-issue: '4230' -Remove MAX_VCPUS_PER_VM in Kconfig
- :acrn-issue: '4232' -Set default KATA_VM_NUM to 1 for SDC
- :acrn-issue: '4247' -[acrn-configuration-tool] Generate Scenario for VM0 communites with VM1 is incorrect.
- :acrn-issue: '4249' -[acrn-configuration-tool]Generated Launchscript but WebUI prompt error msg after we just select passthru-devices:audio_codec
- :acrn-issue: '4255' -[acrn-configuration-tool][nuc7i7dnb][sdc]uos has no ip address
- :acrn-issue: '4260' -[Community][External]webcam switch between 2 UOS.
- :acrn-issue: '4286' -[acrn-configuration-tool] Remove VM1.vcpu_affinity.pcuid=3 for VM1 in sdc scenario

Known Issues
************

> TBD

Reviewer: If no Known Issues exist, please change this to None (instead of TBD).

Author: Waiting for final clear linux version.

Change Log
**********

These commits have been added to the acrn-hypervisor repo since the v1.4
release in Nov 2019 (click on the CommitID link to see details):

.. comment

   This list is obtained from this git command (update the date to pick up
   changes since the last release):

   git log --pretty=format:'- :acrn-commit:`%h` - %s' --after="2020-01-02"

>    - :acrn-commit:'ee74737f' - HV: search rsdp from e820 acpi reclaim region

Reviewer: Does this commit line need to be left-aligned?

- :acrn-commit:'be6c6851' - acrn-config: refine mount device for virtio-blk
- :acrn-commit:'686d7763' - HV: Remove INIT signal notification related code
- :acrn-commit:'d7eb14c5' - HV: Use NMI to replace INIT signal for lapic-pt VMs S5
- :acrn-commit:'29b7aff5' - HV: Use NMI-window exiting to address req missing issue
- :acrn-commit:'d26d8bec' - HV: Don't make NMI injection req when notifying vCPU
- :acrn-commit:'24c2c0ec' - HV: Use NMI to kick lapic-pt vCPU's thread
- :acrn-commit:'23422713' - acrn-config: add 'tap_' perfix for virtio-net
- :acrn-commit:'6383394b' - acrn-config: enable log_setting in all vm
- :acrn-commit:'0b44d64d' - acrn-config: check pass-thruogh device for audio/audio_codec
- :acrn-commit:'75ca1694' - acrn-config: correct vuart1 setting in scenario config
- :acrn-commit:'d52b45c1' - hv:fix crash issue when handling HC_NOTIFY_REQUEST_FINISH
- :acrn-commit:'78139b95' - HV: kconfig: add range check for memory setting
- :acrn-commit:'24994703' - HV: Kconfig: set default Kata num to 1 in SDC
- :acrn-commit:'9d5e72e9' - hv: add lock for ept add/modify/del
- :acrn-commit:'98b3dd94' - acrn-config: set HV_RAM_START above 256M for new board

> - :acrn-commit:'46463900000000' - acrn-config: add 'ramdisk_mod' item tag for tgl-rvp

Reviewer: Does this commit number need to be fixed (or removed)? Sometimes these strange numbers mess up the Release Notes page.

Author: Done!
